infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -394,6 +394,9 @@ private void processSnapshotStart(LogReplication.LogReplicationEntryMsg entry) {
         // Signal start of snapshot sync to the writer, so data can be cleared (on old snapshot syncs)
         snapshotWriter.reset(topologyId, timestamp);
 
+        // Signal the snapshot writer to collect streams for replication with local writes
+        snapshotWriter.addLocalStreamsToClear();
```

Comment:
why do we need to compute this set at the beginning of snapshot sync start?  why cant it be computed in 'clearLocalStreams'? in the apply phase?